packages/flutter_tools/templates/list_detail_app/lib/src/settings/settings_view.dart.tmpl

```diff
@@ -0,0 +1,50 @@
+import 'package:flutter/material.dart';
+import 'package:{{projectName}}/src/settings/settings_controller.dart';
+
+/// Displays the various settings that can be customized by the user.
+///
+/// When a user changes a setting, the SettingsController is updated and
+/// Widgets that listen to the SettingsController are rebuilt.
+class SettingsView extends StatelessWidget {
+  const SettingsView({Key key, @required this.controller}) : super(key: key);
+
+  static const routeName = '/settings';
+
+  final SettingsController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Settings'),
+      ),
+      body: Padding(
+        padding: const EdgeInsets.all(16),
+        // Glue the SettingsController to the theme selection DropdownButton.
+        //
+        // When a user selects a theme from the dropdown list, the
+        // SettingsController is updated, which rebuilds the MaterialApp.
+        child: DropdownButton<ThemeMode>(
+          // Read the selected themeMode from the controller
+          value: controller.themeMode,
+          // Call the updateThemeMode method any time the user selects a theme.
+          onChanged: controller.updateThemeMode,
+          items: const [
+            DropdownMenuItem(
+              value: ThemeMode.system,
+              child: Text('System Theme'),
+            ),
+            DropdownMenuItem(
+              value: ThemeMode.light,
+              child: Text('Light Theme'),
+            ),
+            DropdownMenuItem(
+              value: ThemeMode.dark,
+              child: Text('Dark Theme'),
+            )
+          ],
+        ),
+      ),
+    );
+  }
+}
```

Comment:
no restoration? showing how to do restoration well seems very valuable for a best practices template

Reply:
Are user settings generally a good candidate for State Restoration? In my experience, it's better to use a proper persistence layer backed up by a local database (shared_preferences, sqflite, hive, etc) or a remote source (REST, GraphQL, etc).

That way the user's preferred theme is properly persisted when the user closes the application or restarts the operating system.

In my view, State Restoration is more appropriate for things like Navigation Stack or Forms that need to be restored when the user puts an app into the background then wants to come back to the app.

What do you think?

Reply:
i agree it wouldn't make sense for this particular setting, but I meant for the routes themselves. when the user is on the settings page, then switches to another app, then comes back, without restoration, there's a risk they'll end up on the main page again.

Reply:
Good call. I've added Route restoration!